prefect_kubernetes/worker.py

```diff
@@ -0,0 +1,671 @@
+        flow_run: "FlowRun",
+        configuration: KubernetesWorkerJobConfiguration,
+        task_status: Optional[anyio.abc.TaskStatus] = None,
+    ) -> KubernetesWorkerResult:
+        """
+        Executes a flow run within a Kubernetes Job and waits for the flow run
+        to complete.
+
+        Args:
+            flow_run: The flow run to execute
+            configuration: The configuration to use when executing the flow run.
+            task_status: The task status object for the current flow run. If provided,
+                the task will be marked as started.
+
+        Returns:
+            KubernetesWorkerResult: A result object containing information about the
+                final state of the flow run
+        """
+        with self._get_configured_kubernetes_client(configuration) as client:
+            job = await run_sync_in_worker_thread(
+                self._create_job, configuration, client
+            )
+            pid = await run_sync_in_worker_thread(
+                self._get_infrastructure_pid, job, client
+            )
+            # Indicate that the job has started
+            if task_status is not None:
+                task_status.started(pid)
+
+            # Monitor the job until completion
+            status_code = await run_sync_in_worker_thread(
+                self._watch_job, job.metadata.name, configuration, client
+            )
+            return KubernetesWorkerResult(identifier=pid, status_code=status_code)
+
+    def _get_configured_kubernetes_client(
+        self, configuration: KubernetesWorkerJobConfiguration
+    ) -> "ApiClient":
+        """
+        Returns a configured Kubernetes client.
+        """
+
+        # if a hard-coded cluster config is provided, use it
+        if configuration.cluster_config:
+            return kubernetes.config.new_client_from_config_dict(
+                config_dict=configuration.cluster_config.config,
+                context=configuration.cluster_config.context_name,
+            )
+        else:
+            # If no hard-coded config specified, try to load Kubernetes configuration
+            # within a cluster. If that doesn't work, try to load the configuration
+            # from the local environment, allowing any further ConfigExceptions to
+            # bubble up.
+            try:
+                kubernetes.config.load_incluster_config()
+                config = kubernetes.client.Configuration.get_default_copy()
+                return kubernetes.client.ApiClient(configuration=config)
+            except kubernetes.config.ConfigException:
+                return kubernetes.config.new_client_from_config()
+
+    def _create_job(
+        self, configuration: KubernetesWorkerJobConfiguration, client: "ApiClient"
+    ) -> "V1Job":
+        """
+        Creates a Kubernetes job from a job manifest.
+        """
+        with self._get_batch_client(client) as batch_client:
+            job = batch_client.create_namespaced_job(
```

Comment:
if someone gives us a `job_manifest` that isn't a k8s job, will this throw an exception? If so, would this happen at scheduling time or at execution time?

Reply:
`KubernetesWorkerJobConfiguration._ensure_job_has_compatible_values` will raise an error at execution time for non-job manifests.